src/views/projectExplorer/index.ts

```diff
@@ -71,6 +71,92 @@ export default class ProjectExplorer implements TreeDataProvider<ProjectExplorer
       commands.registerCommand(`vscode-ibmi-projectexplorer.projectExplorer.goToIFSBrowser`, async () => {
         await commands.executeCommand(`ifsBrowser.focus`);
       }),
+      commands.registerCommand(`vscode-ibmi-projectexplorer.projectExplorer.runBuild`, async (element?: Project) => {
+        let iProject: IProject | undefined;
+        if (element && element instanceof Project) {
+          iProject = ProjectManager.get(element.workspaceFolder);
+        } else {
+          iProject = ProjectManager.getActiveProject();
+        }
+
+        if (iProject) {
+          iProject.runBuildOrCompileCommand(true);
+        } else {
+          window.showErrorMessage(l10n.t('Failed to retrieve project'));
+        }
+      }),
+      commands.registerCommand(`vscode-ibmi-projectexplorer.projectExplorer.setBuildCommand`, async (element: Project) => {
+        if (element) {
+          const iProject = ProjectManager.get(element.workspaceFolder);
+
+          if (iProject) {
+            const unresolvedState = await iProject.getUnresolvedState();
+
+            if (unresolvedState) {
+              const command = await window.showInputBox({
+                prompt: l10n.t('Enter build command'),
+                placeHolder: l10n.t('Build command'),
+                value: unresolvedState.buildCommand,
+              });
+
+              if (command) {
+                await iProject.setBuildOrCompileCommand(command, true);
+              }
+            }
+          } else {
+            window.showErrorMessage(l10n.t('Failed to retrieve project'));
+          }
+        }
+      }),
+      commands.registerCommand(`vscode-ibmi-projectexplorer.projectExplorer.runCompile`, async (element?: Project | Uri) => {
+        let iProject: IProject | undefined;
+        if (element) {
+          if (element instanceof Project) {
+            iProject = ProjectManager.get(element.workspaceFolder);
+          } else {
+            iProject = ProjectManager.getProjectFromUri(element);
+          }
+        } else {
+          iProject = ProjectManager.getActiveProject();
+        }
+
+        if (iProject) {
+          iProject.runBuildOrCompileCommand(false);
+        } else {
+          window.showErrorMessage(l10n.t('Failed to retrieve project'));
+        }
+      }),
+      commands.registerCommand(`vscode-ibmi-projectexplorer.projectExplorer.setCompileCommand`, async (element: Project) => {
+        if (element) {
+          const iProject = ProjectManager.get(element.workspaceFolder);
+
+          if (iProject) {
+            const unresolvedState = await iProject.getUnresolvedState();
+
+            if (unresolvedState) {
+              const command = await window.showInputBox({
+                prompt: l10n.t('Enter compile command'),
```

Comment:
definitely need  variable for file name documented here

Reply:
Same as above